### PR TITLE
Update kpod logs to use the new container state and runtime

### PR DESCRIPTION
**Pull request by umohnani8**
Signed-off-by: Urvashi Mohnani <umohnani@redhat.com>

### CODE REVIEW

**Comment by umohnani8** [comment]
@rhatdan @mheon PTAL

**Comment by umohnani8** [comment]
the --tail flag is still broken https://github.com/projectatomic/libpod/issues/16, but the tests pass (not sure why). Will get back to that later.

**Comment by TomSweeneyRedHat** [comment]
LGTM, assuming happy tests.

File: cmd/kpod/logs.go [hunk 1]
```diff
@@ -4,8 +4,9 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/projectatomic/libpod/libpod"
+
```
Comment (rhatdan) [inline cmd/kpod/logs.go]: Remove the extra spaces around the github lines.

File: libpod/runtime_ctr.go [hunk 2]
```diff
@@ -1,6 +1,11 @@
 package libpod
 
 import (
+	"strings"
+	"time"
+
+	"github.com/hpcloud/tail"
+
```
Comment (rhatdan) [inline libpod/runtime_ctr.go]: Remove line

File: libpod/runtime_ctr.go [hunk 3]
```diff
@@ -195,6 +204,49 @@ func (r *Runtime) GetContainers(filters ...ContainerFilter) ([]*Container, error
...
+// log file is created when the container is started/ran
+func (r *Runtime) GetLogs(container string, logChan chan string, opts LogOptions) error {
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+
+	if !r.valid {
+		return ErrRuntimeStopped
+	}
+
+	defer close(logChan)
```
Comment (rhatdan) [inline libpod/runtime_ctr.go]: Should this be before the check above?  Does the user expect the channel to be closed when GetLogs returns?

File: libpod/runtime_ctr.go [hunk 4]
```diff
@@ -159,6 +164,10 @@ func (r *Runtime) LookupContainer(idOrName string) (*Container, error) {
 		return nil, ErrRuntimeStopped
 	}
 
+	return r.lookUpContainer(idOrName)
+}
+
+func (r *Runtime) lookUpContainer(idOrName string) (*Container, error) {
```
Comment (mheon) [inline libpod/runtime_ctr.go]: Why do we need this extra function?

File: libpod/runtime_ctr.go [hunk 5]
```diff
@@ -195,6 +204,49 @@ func (r *Runtime) GetContainers(filters ...ContainerFilter) ([]*Container, error
...
+type LogOptions struct {
+	Details   bool
+	Follow    bool
+	SinceTime time.Time
+	Tail      uint64
+}
+
+// GetLogs returns the logs of a container from the log file
+// log file is created when the container is started/ran
+func (r *Runtime) GetLogs(container string, logChan chan string, opts LogOptions) error {
```
Comment (mheon) [inline libpod/runtime_ctr.go]: Can we move this out to cmd/kpod? I don't think we need anything in the runtime for this aside from the container, which you can just get with LookupContainer

**Comment by mheon** [comment]
I don't like having the log parsing inside the runtime. I think we should move it either into cmd/kpod or into container.go. Otherwise looks fine.

File: cmd/kpod/logs.go (hunk 1 above)
Comment (umohnani8) [reply cmd/kpod/logs.go]: fixed

File: libpod/runtime_ctr.go (hunk 2 above)
Comment (umohnani8) [reply libpod/runtime_ctr.go]: fixed

File: libpod/runtime_ctr.go (hunk 3 above)
Comment (umohnani8) [reply libpod/runtime_ctr.go]: uh, not sure I just modified the code that was already written, which was like this.

File: libpod/runtime_ctr.go (hunk 4 above)
Comment (umohnani8) [reply libpod/runtime_ctr.go]: I was worried about the lock hanging again, but not needed anymore. fixed

File: libpod/runtime_ctr.go (hunk 5 above)
Comment (umohnani8) [reply libpod/runtime_ctr.go]: fixed.

**Comment by mheon** [comment]
LGTM. @rhatdan PTAL

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**